### PR TITLE
Add "Icon" widget to Widget catalog

### PR DESCRIPTION
**Pull request by brianegan**
Replace the Icon Widget's "Placeholder" icon in the Widget Catalog with an icon generated by the assets-for-api-docs tool.

### CODE REVIEW

**Comment by brianegan** [comment]
Hey @Piinks and @Sfshaza -- Emily has asked me to help fill in the Widget catalog! I was planning on using the Api Docs Diagram generator for this purpose and wanted to make sure everyone was Ok with that plan.

Many of the Widgets in the catalog have custom animations or illustrations instead of screenshots of what the Widget looks like on screen, so just wanted to get a gut check if this makes sense or we should keep the "assets-for-api-docs" stuff completely separate from the widget catalog?

File: src/_data/catalog/widgets.json [hunk 1]
```diff
@@ -70,7 +70,7 @@
       "Information displays"
     ],
     "link": "https://api.flutter.dev/flutter/widgets/Icon-class.html",
-    "image": "<img alt='' src='/images/catalog-widget-placeholder.png'>"
+    "image": "<img alt='' src='https://raw.githubusercontent.com/flutter/assets-for-api-docs/master/assets/widgets/icon.png'>"
```
Comment (Piinks) [inline src/_data/catalog/widgets.json]:
```suggestion
    "image": "<img alt='' src='https://flutter.github.io/flutter/assets-for-api-docs/master/assets/widgets/icon.png'>"
```
I would wait and double check that this resolves to the desired asset when it lands in the repo. :)

**Comment by Piinks** [comment]
Hmm.. Why not? It is a useful script that can create diagrams, animations and illustrate interactions with various widgets, *and* anything generated for the website could be repurposed for the docs and vice versa. Sounds like a great idea @brianegan! 
What do you think @Sfshaza? 
I am also curious to see what @gspencergoog thinks. 
My only concern right now would be potential bloat in that repository and confusion between what's being used where.

File: src/_data/catalog/widgets.json (hunk 1 above)
Comment (Piinks) [reply src/_data/catalog/widgets.json]: ingore previous. This is the proper link:
https://flutter.github.io/assets-for-api-docs/assets/widgets/icon.png

Comment (brianegan) [reply src/_data/catalog/widgets.json]: Thanks!

**Comment by sfshaza2** [comment]
I am definitely OK with this! I actually suggested to Emily that you do this..  :)

**Comment by brianegan** [comment]
Sounds great! I thought everyone had been looped in, but just wanted to be sure. Thanks again for your help :)

**Pull request merged**